applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -819,36 +765,32 @@ namespace Kratos
 		ProcessInfo& rCurrentProcessInfo)
 	{
 		KRATOS_TRY;
-		const int number_of_nodes = GetGeometry().PointsNumber();
-		const int dimension = GetGeometry().WorkingSpaceDimension();
-		const unsigned int MatSize = number_of_nodes * dimension * 2;
-
-		if (rMassMatrix.size1() != MatSize) {
-			rMassMatrix.resize(MatSize, MatSize, false);
+		if (rMassMatrix.size1() != element_size) {
+			rMassMatrix.resize(element_size, element_size, false);
 		}
-		rMassMatrix = ZeroMatrix(MatSize, MatSize);
+		rMassMatrix = ZeroMatrix(element_size, element_size);
 
 
 
-		if (this->GetProperties().Has(LUMPED_MASS_MATRIX) == true) {
+		if (this->GetProperties().Has(LUMPED_MASS_MATRIX)) {
 			this->mIsLumpedMassMatrix = GetProperties()[LUMPED_MASS_MATRIX];
 		}
 		else this->mIsLumpedMassMatrix = false;
 
 
 
-		if (this->mIsLumpedMassMatrix == true)
+		if (this->mIsLumpedMassMatrix)
 		{
 			this->CalculateLumpedMassMatrix(rMassMatrix, rCurrentProcessInfo);
 		}
 		else
 		{
 			this->CalculateConsistentMassMatrix(rMassMatrix, rCurrentProcessInfo);
 
-			Matrix RotationMatrix = ZeroMatrix(MatSize);
-			Matrix aux_matrix = ZeroMatrix(MatSize);
+			bounded_matrix<double,element_size,element_size> RotationMatrix = ZeroMatrix(element_size,element_size);

```

Comment:
I would suggest to use array_1d which is faster and is more standard in Kratos

Reply:
as stated in issue #664 the usage of array_1d in combination with cross products needs an inverse input of the arrays. This is why I try to avoid using array_1d

Reply:
Make sense!

Reply:
Now I see, we should move with that issue. For now is ok what you did....